stat/distmv/studentst.go

```diff
@@ -208,18 +206,25 @@ func studentsTConditional(observed []int, values []float64, nu float64, mu []flo
 // findUnob returns the unobserved variables (the complementary set to observed).
 // findUnob panics if any value repeated in observed.
 func findUnob(observed []int, dim int) (unobserved []int) {
-	var setOb intsets.Sparse
-	for _, v := range observed {
-		setOb.Insert(v)
+	all := make([]int, 0, dim)
+	for i := range dim {
+		all = append(all, i)
 	}
-	var setAll intsets.Sparse
-	for i := 0; i < dim; i++ {
-		setAll.Insert(i)
+	for _, obs := range observed {
+		if obs >= 0 && obs < dim {
```

Comment:
nit: we use the convention of left <= right for ranges, so this would be `0 <= obs && obs < dim`.